src/App.tsx

```diff
@@ -1,61 +1,113 @@
+import { ChangeEvent, FormEvent, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const updatedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId) || null,
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(updatedTodos);
+  const [title, setTitle] = useState('');
+  const [userName, setUserName] = useState(0);
+
+  const [isTitleError, setTitleError] = useState(false);
+  const [isUserError, setUserError] = useState(false);
+
+  const newTodoId = Math.max(...todos.map(todo => todo.id));
+
+  const resetForm = () => {
+    setTitle('');
+    setUserName(0);
+  };
+
+  const handleTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUser = (event: ChangeEvent<HTMLSelectElement>) => {
+    setUserName(+event.target.value);
+    setUserError(false);
+  };
+
+  const handleFormSubmit = (event:FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTitleError(!title);
+    setUserError(!userName);
+
+    const newTodo: Todo = {
+      id: newTodoId + 1,
+      userId: userName,
+      title,
+      completed: false,
+      user: getUser(userName),
+    };
+
+    if (title && userName) {
+      setTodos(currentTodos => [...currentTodos, newTodo]);
+      resetForm();
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleFormSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          {'Title: '}
```

Comment:
Wrap it with the label. Check other similar cases.

Reply:
I fixed this bugs.